lib/system/memalloc.nim

```diff
@@ -304,6 +304,80 @@ when hasAlloc and not defined(js):
     ## or other memory may be corrupted.
     deallocShared(p)
 
+  include bitmasks  
+
+  template `+!`(p: pointer, s: SomeInteger): pointer =
+    cast[pointer](cast[int](p) +% s)
+
+  template `-!`(p: pointer, s: SomeInteger): pointer =
+    cast[pointer](cast[int](p) -% s)
+
+  proc alignedAlloc(size, align: Natural): pointer =
+    if align <= MemAlign:
+      when compileOption("threads"):
+        result = allocShared(size)
+      else:
+        result = alloc(size)
+    else: 
+      when compileOption("threads"):
+        let base = allocShared(size + align - 1 + sizeof(uint16))
```

Comment:
Write a comment that explains this logic. And also why it doesn't produce **un**aligned stores (which can be performance desasters on non-x86 CPUs).

Reply:
I have added the comments. I have carefully reviewed the code for unaligned reads or stores. I dont' see any.